src/controllers/mainController.ts

```diff
@@ -220,9 +220,17 @@ export default class MainController implements vscode.Disposable {
             this._event.on(Constants.cmdClearPooledConnections, async () => {
                 await this.onClearPooledConnections();
             });
-            this.registerCommand(Constants.cmdDeployNewDatabase);
-            this._event.on(Constants.cmdDeployNewDatabase, () => {
-                this.onDeployNewDatabase();
+            this.registerCommandWithArgs(Constants.cmdDeployNewDatabase);
+            this._event.on(Constants.cmdDeployNewDatabase, (args?: any) => {
+                let initialConnectionGroup: { id?: string } | undefined = undefined;
```

Comment:
I think this can just be `initialConnectionGroupId: string` instead, if all that' the rest of this is doing is extracting the ID.